Portal/View/UIKit/Renderers/TextFieldRenderer.swift

```diff
@@ -18,63 +20,84 @@ internal struct TextFieldRenderer<MessageType, RouteType: Route>: UIKitRenderer
-        style.textAligment                        |> { self.textAlignment = $0.asNSTextAligment }
+    internal func apply<MessageType>(
+        changeSet: TextFieldChangeSet<MessageType>,
+        layoutEngine: LayoutEngine) -> Render<MessageType> {
+        
+        apply(changeSet: changeSet.properties)
+        apply(changeSet: changeSet.baseStyleSheet)
+        apply(changeSet: changeSet.textFieldStyleSheet)
+        layoutEngine.apply(changeSet: changeSet.layout, to: self)
+        
+        return Render(view: self, mailbox: getMailbox(), executeAfterLayout: .none)
     }
     
 }
 
-extension UITextField {
+fileprivate extension UITextField {
     
-    fileprivate func dispatch<MessageType>(
-        message: MessageType,
-        for event: UIControlEvents,
-        with mailbox: Mailbox<MessageType> = Mailbox()) -> Mailbox<MessageType> {
-        
-        let dispatcher = MessageDispatcher(mailbox: mailbox, message: message)
-        self.register(dispatcher: dispatcher)
-        self.addTarget(dispatcher, action: dispatcher.selector, for: event)
-        return dispatcher.mailbox
+    fileprivate func apply<MessageType>(changeSet: [TextFieldProperties<MessageType>.Property]) {
+        for propertie in changeSet {
+            switch propertie {
+                
+            case .text(let text):
+                self.text = text
+            
+            case .placeholder(let placeholder):
+                self.placeholder = placeholder
+                
+            case .onEvents(let events):
+                apply(events: events)
+            }
+        }
+    }
+    
+    fileprivate func apply<MessageType>(events: TextFieldEvents<MessageType>) {
+        for (event, maybeEvent) in getMessagesByEvent(events: events) {
+            if let message = maybeEvent {
+                _ = self.on(event: event, dispatch: message)
+            } else {
+                _ = self.unregisterDispatcher(for: event) as MessageDispatcher<MessageType>?
+            }
+        }
+    }
+    
+    fileprivate func apply(changeSet: [TextFieldStyleSheet.Property]) {
+        for propertie in changeSet {
+            switch propertie {
+                
+            case .textAligment(let aligment):
+                self.textAlignment = aligment.asNSTextAligment
+            
+            case .textColor(let color):
+                self.textColor = color.asUIColor
+                
+            case .textFont(let font):
+                let fontSize = self.font?.pointSize ?? CGFloat(defaultTextFieldFontSize)
+                self.font = font.uiFont(withSize: fontSize)
+                
+            case .textSize(let textSize):
+                let fontName = self.font?.fontName
+                fontName |> { self.font = UIFont(name: $0, size: CGFloat(textSize)) }
+            }
+        }
+    }
+    
+    fileprivate func getMessagesByEvent<MessageType>(events: TextFieldEvents<MessageType>) -> [(UIControlEvents, MessageType?)] {
```

Comment:
Just a small detail, but what I was suggesting before was to add an extension to `TextFieldEvents` inside `TextFieldRenderer.swift` file.
```swift
fileprivate extension TextFieldEvents {

  fileprivate var messagesByEvent: [(UIControlEvents, MessageType?)] {
    return [
            (.editingDidBegin, events.onEditingBegin),
            (.editingChanged, events.onEditingChanged),
            (.editingDidEnd, events.onEditingEnd)
        ]
  }

}
```